lib/cuffsert/rxcfclient.rb

```diff
@@ -57,8 +58,12 @@ def update_stack(stack_id, change_set_id)
       Rx::Observable.create do |observer|
         start_time = record_start_time
         @cf.execute_change_set(change_set_name: change_set_id)
-        stack_events(stack_id, start_time) do |event|
-          observer.on_next(event)
+        begin
+          stack_events(stack_id, start_time) do |event|
+            observer.on_next(event)
+          end
+        rescue => e
+          observer.on_error(e)
```

Comment:
Given that this calls `exit`, this will raise an error that means that `on_completed` will not be called. If that is intentional, I think it would be nicer to stick the on_completed call in an `else` block (or inside the `begin` block depending on the whether errors in the call to `on_completed` can be meaningfully handled using `on_error`.

Reply:
Ah, good catch. The Rx contract is to call `on_error` or `on_completed` but not both. RxRuby has some guards that will ensure that `on_completed` will be a no-op in the error case, but you are absolute right that the code should not depend on that. Fixed.